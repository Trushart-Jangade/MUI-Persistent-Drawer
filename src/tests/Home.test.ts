import * as React from 'react';
import * as ReactShallowRenderer from 'react-test-renderer/shallow';
import Home from '../Home';
import { shallow } from 'enzyme';
// const ReactShallowRenderer = require('react-test-renderer/shallow');
test('should render Home correctly', () => {
    const renderer = ReactShallowRenderer.createRenderer();
    renderer.render(React.createElement(Home));
    // console.log(renderer.getRenderOutput());
    
    expect(renderer.getRenderOutput()).toMatchSnapshot();
});

test('should have H1 tag', () => {
    const h1 = shallow(React.createElement(Home));
  
    expect(h1.find('h1').text()).toEqual('Home Component');
    
    expect(h1.find('div').containsMatchingElement(React.createElement('h1',{children:'Home Component'}))).toEqual(true);
  
    // Snapshot demo
    expect(Home).toMatchSnapshot();
  });