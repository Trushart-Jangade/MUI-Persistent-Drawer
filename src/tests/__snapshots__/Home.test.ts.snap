// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should have H1 tag 1`] = `[Function]`;

exports[`should render Home correctly 1`] = `
<div>
  <h1>
    Home Component
  </h1>
</div>
`;
